<launch>
  <arg name="SIMULATION_MODE" default="false"/>
  <arg name="REAL_MODE" default="false"/>
  <arg name="ROSBAG_MODE" default="false"/>
  <arg name="JAXON" default="true"/>
  <arg name="HRP2" default="false"/>

  <!-- static tf "car_center" (parent link: utorso) publisher for gazebo_drive_simulator -->
  <node if="$(arg SIMULATION_MODE)" ns="/drive/recognition" pkg="tf" type="static_transform_publisher" name="TF_car_center" args="0.05 0.3 -1.2 0 0 0 utorso car_center 100" /> <!-- initial position is y : -0.6, z : -0.1 // start_gazebo_drive_shimulator.sh -->

  <!-- static tf "car_center" (parent link: multisense/head_root) publisher for ROSbags -->
  <include if="$(arg ROSBAG_MODE)" file="$(find drc_task_common)/launch/fc/vehicle/multisense_rosbag_static_tf_publisher.launch" />

  <group if="$(arg REAL_MODE)">
    <group ns="/drive/recognition">
      <!-- dynamic_tf_publisher -->
      <node pkg="dynamic_tf_publisher" type="tf_publish.py" name="marker_dynamic_tf_publisher" respawn="true"/>
      <!-- checkerboard_detector -->
      <param name="display" type="int" value="0"/>
      <param name="rect0_size_x" type="double" value="0.0152"/>
      <param name="rect0_size_y" type="double" value="0.0152"/>
      <param name="grid0_size_x" type="int" value="3"/>
      <param name="grid0_size_y" type="int" value="5"/>
      <param name="board_type" value="acircle" />
      <param name="use_P" value="true" />
      <param name="invert_color" value="true" />
      <param name="message_throttle" value="5" />
      <param name="type0" type="string" value="data/ricebox.kinbody.xml"/>
      <node pkg="checkerboard_detector" name="checkerboard_detector" type="checkerboard_detector" respawn="true">
        <remap from="camera_info" to="/multisense/left/camera_info"/>
        <remap from="image" to="/multisense/left/image_rect_color"/>
        <remap from="objectdetection_pose" to="marker/car_pose" />
      </node>
      <group if="$(arg JAXON)">
        <!-- tf "car_marker" publisher from PoseStamped -->
        <node pkg="roseus" type="roseus" name="jaxon_TF_BODY_marker" respawn="true"
              args="$(find drc_task_common)/euslisp/vehicle/jaxon-car_center-publisher.l" >
          <remap from="car_pose_marker" to="marker/car_pose" />
        </node>
        <!-- static tf "car_center" (parent link: car_marker) publisher -->
        <node pkg="tf" type="static_transform_publisher" name="jaxon_TF_marker_car_center" respawn="true" args="1.323766 0.0 0.1942 3.141593 -1.082 -3.141593 car_marker car_center 100" />
      </group>
      <group if="$(arg HRP2)">
        <node  pkg="roseus" type="roseus" name="hrp2_TF_BODY_marker" respawn="true"
               args="$(find drc_task_common)/euslisp/vehicle/hrp2-car_center-publisher.l" >
          <remap from="car_pose_marker" to="marker/car_pose" />
        </node>
        <!-- static tf "car_center" (parent link: car_marker) publisher -->
        <node pkg="tf" type="static_transform_publisher" name="hrp2_TF_marker_car_center" respawn="true" args="1.323766 0.0 0.1942 3.141593 -1.07 -3.141593 car_marker car_center 100" />
      </group>
    </group>
  </group>
</launch>
